stubs/tensorflow/tensorflow/_aliases.pyi

```diff
@@ -0,0 +1,36 @@
+# Collection of commonly need type aliases. These are all private
+# and do not exist at runtime.
+
+from typing import Iterable, Mapping, Optional, Sequence, TypeVar, Union
+
+import numpy as np
+import tensorflow as tf
+
+# These aliases mostly ignore rank/shape/dtype information as that
+# will complicate the types heavily and can be a follow up problem.
+_FloatDataSequence = Union[Sequence[float], Sequence["_FloatDataSequence"]]
```

Comment:
Several of these type aliases are recursive. What would be recommended way to write them? Arbitrarily nested sequences or json like containers (nested maps/sequences) are very common types needed. Should I leave them like this, use a recursive protocol, avoid recursion and fallback to Any beyond given depth, or something else?

_ContainerGeneric and _DataSequence alises are the main recursive ones here.